packages/Modal/src/index.tsx

```diff
@@ -66,25 +66,26 @@ const ModalComponent = forwardRef<'div', ModalProps>(
     },
     ref
   ) => {
-    const closeModal = () => {
+    const prevHide = useMemo(() => state.hide, [state])
+
+    state.hide = useCallback(() => {
       onClose?.()
-      state.hide()
-    }
+      prevHide()
+    }, [prevHide, onClose])
 
     return (
       <ModalProvider state={state}>
         <S.Backdrop {...state} hideOnClickOutside={hideOnClickOutside}>
           <S.Dialog
             aria-label={ariaLabel}
-            hide={closeModal}
             hideOnClickOutside={hideOnClickOutside}
             preventBodyScroll={false}
             ref={ref}
             size={size}
             {...state}
             {...rest}
           >
-            <CloseElement onClick={closeModal} />
+            <CloseElement onClick={() => state.hide()} />
```

Comment:
```suggestion
            <CloseElement onClick={state.hide} />
```

Reply:
It triggers an eslint error "Handler function for onClick prop key must be a camelCase name beginning with 'handle' only"